pkg/diskmaker/controllers/lvset/reconcile.go

```diff
@@ -0,0 +1,265 @@
+	"github.com/openshift/local-storage-operator/pkg/controller/nodedaemon"
+	"github.com/openshift/local-storage-operator/pkg/internal"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/types"
+	corev1helper "k8s.io/kubernetes/pkg/apis/core/v1/helper"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	staticProvisioner "sigs.k8s.io/sig-storage-local-static-provisioner/pkg/common"
+)
+
+// Reconcile reads that state of the cluster for a LocalVolumeSet object and makes changes based on the state read
+// and what is in the LocalVolumeSet.Spec
+// TODO(user): Modify this Reconcile function to implement your Controller logic.  This example creates
+// a Pod as an example
+// Note:
+// The Controller will requeue the Request to be processed again if the returned error is non-nil or
+// Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
+func (r *ReconcileLocalVolumeSet) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	reqLogger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
+	reqLogger.Info("Reconciling LocalVolumeSet")
+
+	// Fetch the LocalVolumeSet instance
+	instance := &localv1alpha1.LocalVolumeSet{}
+	err := r.client.Get(context.TODO(), request.NamespacedName, instance)
+	if err != nil {
+		if kerrors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, err
+	}
+
+	// get node name
+	nodeName := getNodeNameEnvVar()
+	// get node labels
+	node := &corev1.Node{}
+	err = r.client.Get(context.TODO(), types.NamespacedName{Name: nodeName}, node)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	// ignore LocalVolmeSets whose LabelSelector doesn't match this node
+	// NodeSelectorTerms.MatchExpressions are ORed
+	matches, err := nodeSelectorMatchesNodeLabels(node, instance.Spec.NodeSelector)
+	if err != nil {
+		reqLogger.Error(err, "failed to match nodeSelector to node labels")
+		return reconcile.Result{}, err
+	}
+
+	if !matches {
+		return reconcile.Result{}, nil
+	}
+
+	storageClass := instance.Spec.StorageClassName
+
+	// get associated config
+	cm := &corev1.ConfigMap{}
+	// TOODO(rohan) export/import name
+	err = r.client.Get(context.TODO(), types.NamespacedName{Name: nodedaemon.ProvisionerConfigMapName, Namespace: request.Namespace}, cm)
+	if err != nil {
+		reqLogger.Error(err, "could not get provisioner configmap")
+		return reconcile.Result{}, err
+	}
+
+	provisionerConfig := staticProvisioner.ProvisionerConfiguration{}
+	staticProvisioner.ConfigMapDataToVolumeConfig(cm.Data, &provisionerConfig)
```

Comment:
lets move the device discovery and symlinking code into its own function rather than having everything here.

Reply:
fixed